packages/story-editor/src/components/floatingMenu/elements/toggleBold.js

```diff
@@ -19,18 +19,47 @@
  */
 import { Icons } from '@googleforcreators/design-system';
 import { __ } from '@googleforcreators/i18n';
+import { trackEvent } from '@googleforcreators/tracking';
+import { useCallback } from '@googleforcreators/react';
 
 /**
  * Internal dependencies
  */
-import { IconButton } from './shared';
+import useRichTextFormatting from '../../panels/design/textStyle/useRichTextFormatting';
+import updateProperties from '../../inspector/design/updateProperties';
+import { useStory } from '../../../app';
+import { IconButton, useProperties } from './shared';
 
 function ToggleBold() {
+  const { content } = useProperties(['content']);
+  const updateSelectedElements = useStory(
+    (state) => state.actions.updateSelectedElements
+  );
+
+  const pushUpdate = useCallback(
+    (update) => {
+      trackEvent('floating_menu', {
+        name: 'set_bold',
+        element: 'text',
+      });
+      updateSelectedElements({
+        properties: (element) => updateProperties(element, update, true),
+      });
+    },
+    [updateSelectedElements]
+  );
+  const {
+    textInfo: { isBold },
+    handlers: { handleClickBold },
+  } = useRichTextFormatting([{ content, type: 'text' }], pushUpdate);
+
   return (
     <IconButton
+      isToggled={isBold}
       Icon={Icons.LetterBBold}
-      title={__('Toggle bold text', 'web-stories')}
-      onClick={() => {}}
+      title={__('Toggle bold', 'web-stories')}
+      onClick={() => handleClickBold(!isBold)}
+      tabIndex="0"
```

Comment:
FYI: Added this for now since it was impossible to get into the toggles otherwise using keyboard -- some of the menu items seem to only allow tabbing and some using arrow keys. This can be adjusted in #10872.